Dima/netflix-roulette/src/Pages/Home/Home.tsx

```diff
@@ -30,84 +30,80 @@ const connector = connect(mapStateToProps, mapDispatchToProps)
+function Home(props: PropsFromRouteAndRedux) {
 
-  componentDidMount() {
-    const URLData = parse(this.props.location.search) as { search: string, searchBy: string };
+  useEffect(() => {
+    const URLData = parse(props.location.search) as { search: string, searchBy: string };
     const { searchBy, search, } = URLData
-    this.props.filmsRequested(0, 'release_date', searchBy, search, false);
-  };
-  componentDidUpdate(prevProps: PropsFromRouteAndRedux) {
-    if(this.props.location !== prevProps.location && this.props.history.action !== 'PUSH') {
-      const URLData = parse(this.props.location.search) as { 
-          search: string, 
-          searchBy: string, 
-          sortBy: string 
-        };
-        const { search, searchBy, sortBy } = URLData
-        console.log(sortBy)
-        this.props.filmsRequested(0, sortBy, searchBy, search, false);
-    };
-  };
+    props.filmsRequested({offset: 0, sortBy: 'release_date', searchBy, search, pagination: false});
+  }, []);
+  useEffect(() => {
+    if(props.history.action === 'PUSH') {
+      const URLData = parse(props.location.search) as { 
+        search: string, 
+        searchBy: string, 
+        sortBy: string 
+      };
+      const { search, searchBy, sortBy } = URLData
+      props.filmsRequested({offset: 0, sortBy, searchBy, search, pagination: false})
+    }
+  }, [props.location]);
 
-  handleSearchChangeForSearch = ( { search, searchBy }: { search: string, searchBy: string } ) => {
-    const URLData = parse(this.props.location.search) as { sortBy: string, search: string, searchBy: string };
+  const handleSearchChangeForSearch = ( { search, searchBy }: { search: string, searchBy: string } ) => {
+    const URLData = parse(props.location.search) as { sortBy: string, search: string, searchBy: string };
     const { sortBy } = URLData
     let requestData = stringify({ ...URLData, search, searchBy })
-    this.props.filmsRequested(0, sortBy, searchBy, search)
-    this.props.history.push({
-      pathname: '/search/',
+    props.filmsRequested({offset: 0, sortBy, searchBy, search})
+    props.history.push({
+      pathname: '/search',
       search: requestData
     });
 
   };
 
-  handleSearchChangeForSort = ({ sortBy }: { sortBy: string }) => {
-    const URLData = parse(this.props.location.search) as { search: string, searchBy: string };
+  const handleSearchChangeForSort = ({ sortBy }: { sortBy: string }) => {
+    const URLData = parse(props.location.search) as { search: string, searchBy: string };
     const { searchBy, search } = URLData
     let requestData = stringify({ ...URLData, sortBy } )
-    console.log(requestData)
-    this.props.filmsRequested(0, sortBy, searchBy, search)
-    this.props.history.push( {
-      pathname: '/search/',
+    props.filmsRequested({offset: 0, sortBy, searchBy, search})
+    props.history.push( {
+      pathname: '/search',
       search: requestData
     });
   };
```

Comment:
You duplicate your code a lot. Try to move repeatable logic into reusable methods.